packages/openapi-code-generator/src/typescript/common/schema-builders/joi-schema-builder.ts

```diff
@@ -108,6 +108,14 @@ export class JoiBuilder extends AbstractSchemaBuilder<
       .join(".")
   }
 
+  protected transform(
+    schema: string,
+    transformation: string | ((it: unknown) => unknown),
+  ) {
+    // TODO: is it possible to do arbitrary transformations with `joi`?
```

Comment:
It looks like the way to do this with `joi` would be to register the transformation as an extension like shown in https://github.com/hapijs/joi/issues/2796#issuecomment-1236430375

This will pretty much require that we make the functions nameable / reusable as we wouldn't want to register an extension for the same thing over and over.